include/dds/core/xtypes/AnnotKind.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef OMG_DDS_CORE_ANNOTKIND_H
+#define OMG_DDS_CORE_ANNOTKIND_H
```

Comment:
as discussed, since i was using the Type into detail/Annotation, in order to avoid circular inclusion i detached the annotationkind definition in a separated file.